test/Microsoft.ML.Tests/Transformers/CategoricalHashTests.cs

```diff
@@ -0,0 +1,300 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Data.IO;
+using Microsoft.ML.Runtime.Model;
+using Microsoft.ML.Runtime.RunTests;
+using Microsoft.ML.Runtime.Tools;
+using System;
+using System.IO;
+using System.Linq;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.ML.Tests.Transformers
+{
+    public class CategoricalHashTests : TestDataPipeBase
+    {
+        public CategoricalHashTests(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        private class TestClass
+        {
+            public string A;
+            public string B;
+            public string C;
+        }
+
+        private class TestMeta
+        {
+            [VectorType(2)]
+            public string[] A;
+            public string B;
+            [VectorType(2)]
+            public float[] C;
+            public float D;
+            [VectorType(2)]
+            public string[] E;
+            public string F;
+        }
+
+        [Fact]
+        public void CategoricalHashWorkout()
+        {
+            var data = new[] { new TestClass() { A = "1", B = "2", C = "3", }, new TestClass() { A = "4", B = "5", C = "6" } };
+
+            var dataView = ComponentCreation.CreateDataView(Env, data);
+            var pipe = new CategoricalHashEstimator(Env, new[]{
+                    new CategoricalHashEstimator.ColumnInfo("A", "CatA", CategoricalTransform.OutputKind.Bag),
+                    new CategoricalHashEstimator.ColumnInfo("A", "CatB", CategoricalTransform.OutputKind.Bin),
+                    new CategoricalHashEstimator.ColumnInfo("A", "CatC", CategoricalTransform.OutputKind.Ind),
+                    new CategoricalHashEstimator.ColumnInfo("A", "CatD", CategoricalTransform.OutputKind.Key),
+                });
+
+            TestEstimatorCore(pipe, dataView);
+            Done();
+        }
+
+        [Fact]
+        public void CategoricalHashStatic()
```

Comment:
>CategoricalHashStatic [](start = 20, length = 21)

move to static pipeline tests assembly

Reply:
Can you create separate issue, since we have bunch (probably most of them)  in Tests now?

---
In reply to: [220399916](https://github.com/dotnet/machinelearning/pull/1033#discussion_r220399916) [](ancestors = 220399916)